File: src/main/java/org/theosib/Utils/FileLocator.java
package org.theosib.Utils;

import java.io.File;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public class FileLocator {
    public static enum FileCategory {
        Textures("textures"),
        Blocks("blocks"),
        Shader("shaders"),
        Chunk("chunks"),
        Config("config");

        public final String subdir;

        private FileCategory(String s) {
            this.subdir = s;
        }
    }

    static String baseDir = null;
    public static void setBaseDir(String bd) {
        baseDir = bd;
    }

    public static String computePath(FileCategory category, String filename) {
        StringBuilder sb = new StringBuilder();
        sb.append(baseDir).append(File.separator).append(category.subdir);
        if (filename == null) return sb.toString();
        return sb.append(File.separator).append(filename).toString();
    }

    public static String getFileAsString(FileCategory category, String filename) throws IOException {
        String path = computePath(category, filename);
        return Files.readString(Path.of(path));
    }
}

File: junk/FileLocator.scala
package org.theosib.Files

import scala.reflect.io.File


trait FileCategory {
  def subdir: String;

  case class Texture() extends FileCategory {
    override def subdir: String = "textures"
  }

  case class Mesh() extends FileCategory {
    override def subdir: String = "blocks"
  }

  case class Shader() extends FileCategory {
    override def subdir: String = "shaders"
  }

  case class Chunk() extends FileCategory {
    override def subdir: String = "chunks"
  }

  case class Config() extends FileCategory {
    override def subdir: String = "config"
  }
}



object FileLocator {
  var base_dir: String = null

  def setBaseDir(base: String): Unit = {
    base_dir = base
  }

  def computePath(category: FileCategory, filename: String): String = {
    val sb = new StringBuilder()
    sb.append(base_dir).append(File.pathSeparator).append(category.subdir)
    if (filename == null) return sb.toString()
    sb.append(File.pathSeparator).(filename).toString()
  }
}
